dhall/src/Dhall/TypeCheck.hs

```diff
@@ -2677,6 +2689,55 @@ prettyTypeMessage (ListAppendMismatch expr0 expr1) = ErrorMessages {..}
         txt0 = insert expr0
         txt1 = insert expr1
 
+prettyTypeMessage (MustCompleteARecord expr0 expr1) = ErrorMessages {..} 
+ where
+   short = "You can only complete records" 
+
+   long = 
+        "Explanation: You can only complete records using the ❰::❱ operator:             \n\
+        \                                                                                \n\
+        \    ┌─────────────────────────────────────────────────────────────────────────┐ \n\
+        \    │ {Type = {foo : Bool, bar : Natural}, default = {bar = 2}::{foo = True}} │ \n\
+        \    └─────────────────────────────────────────────────────────────────────────┘ \n\
+        \                                                                                \n\
+        \... The left-hand side of :: must be a record with 'Type' and 'default' keys    \n\
+        \                                                                                \n\
+        \You tried to record complete the following value:                               \n\
+        \                                                                                \n\
+        \" <> txt0 <> "\n\
+        \                                                                                \n\
+        \... which is not a record. It is:                                               \n\
+        \                                                                                \n\
+        \" <> txt1 <> "\n"
+      where
+        txt0 = insert expr0
+        txt1 = insert expr1
+
+prettyTypeMessage (InvalidRecordCompletion fieldName expr0) = ErrorMessages {..} 
+ where
+   short = "Completion record is missing a field: " <> pretty fieldName
```

Comment:
@Gabriel439 [used](https://github.com/dhall-lang/dhall-kubernetes/pull/84) the name "schema" for these "completion records" – not sure whether we should adopt it here.

Reply:
Would 'completion schema' be ok?

Reply:
Seems good to me! :+1: